src/js/tech/flash.js

```diff
@@ -438,13 +448,19 @@ Flash.onEvent = function(swfID, eventName){
 Flash.onError = function(swfID, err){
   const tech = Dom.getEl(swfID).tech;
   const msg = 'FLASH: '+err;
+  const errorObj = {
+    message: msg
+  };
+
+  tech.error_ = errorObj;
 
   if (err === 'srcnotfound') {
-    tech.trigger('error', { code: 4, message: msg });
+    errorObj.code = 4;
+    tech.trigger('error', errorObj);
 
   // errors we haven't categorized into the media errors
   } else {
-    tech.trigger('error', msg);
+    tech.trigger('error', errorObj);
```

Comment:
`tech.trigger('error', errorObj);` is now the same in both places, so it could be moved outside the conditional.